### PR TITLE
Replacing Travis with Github actions

### LINKED ISSUE #1039: dev: use github actions
It's time to use git actions for things like goreleaser, should be more maintainable.

### PR DESCRIPTION
**Pull request by iwankgb**
Important notes:
* I had to update `go.mod` and `go.sum` in `tools/` - I wasn't able to run `make generate` otherwise
* goreleaser does not support `gomips` key in the configuration
* I got rid of `time` in `Makefile` to support builds on Windows.

I decided to extend builds to Windows and MacOS but:
* tests on Windows fail miserably
* tests on MacOS timeout on Go 1.14 (sometimes)

Release build was tested for [Docker](https://hub.docker.com/r/iwankgb/golangci-lint/tags) and [Github releases](https://github.com/iwankgb/golangci-lint/releases/tag/v2.6). I was not able to test Homebrew. Secrets must be set for hub.docker.com (`GOLANGCI_LINT_DOCKER_TOKEN`) and Github (`GOLANGCI_LINT_TOKEN`) authentication

Fixes #1039, supersedes #1054

### CODE REVIEW

**Comment by iwankgb** [comment]
@ernado you may need to check some configuration options for the repository to enable Github Actions.

**Comment by ernado** [comment]
Oh, great job! 

Can you please leave Travis for deploy script?
We are running goreleaser on version tag and I'm not sure that porting to actions is trivial.

**Comment by iwankgb** [comment]
I have already ported and tested it - see PR description.

**Comment by ernado** [comment]
Oh, now I see.

@jirfag PTAL, seems like we should add secrets in settings.
https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets

**Comment by iwankgb** [comment]
There will be one more for Homebrew but I have no experience with the tool and I have no idea how to authenticate.

File: .goreleaser.yml [hunk 1]
```diff
@@ -25,8 +25,6 @@ builds:
     goarm:
       - 6
       - 7
-    gomips:
```
Comment (jirfag) [inline .goreleaser.yml:28]: what's wrong with that?

**Comment by jirfag** [comment]
I can fix and check `brew` myself

**Pull request merged**